core/src/test/java/org/apache/iceberg/rest/requests/TestCommitTransactionRequestParser.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.requests;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.iceberg.MetadataUpdate;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.rest.requests.UpdateTableRequest.UpdateRequirement;
+import org.junit.jupiter.api.Test;
+
+public class TestCommitTransactionRequestParser {
+
+  @Test
+  public void nullAndEmptyCheck() {
+    assertThatThrownBy(() -> CommitTransactionRequestParser.toJson(null))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid commit tx request: null");
+
+    assertThatThrownBy(() -> CommitTransactionRequestParser.fromJson((JsonNode) null))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse commit tx request from null object");
+
+    assertThatThrownBy(() -> CommitTransactionRequestParser.fromJson("{}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing field: table-changes");
+
+    assertThatThrownBy(() -> CommitTransactionRequestParser.fromJson("{\"table-changes\":{}}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse commit tx request from non-array: {}");
+
+    assertThatThrownBy(() -> CommitTransactionRequestParser.fromJson("{\"table-changes\":[]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid table changes: empty");
+  }
+
+  @Test
+  public void invalidTableIdentifier() {
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"ns1.table1\" : \"ns1.table1\"}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing field: identifier");
```

Comment:
I think this is okay. Each table change in a transaction must specify the table, but I would rather this be a validation in `CommitTransactionRequest` because it isn't really a parser issue, right? If this is delegated to a `UpdateTableRequest` parser, that parser would allow requests missing the `identifier` field.